tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge_wrapper/signup_info_kme.cpp

```diff
@@ -60,16 +69,48 @@ std::map<std::string, std::string> CreateEnclaveDataKME(
     result["enclave_quote"] = enclave_quote;
 
     return result;
-}
+}  // SignupInfoKME::CreateEnclaveData
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+std::map<std::string, std::string> SignupInfoKME::UnsealEnclaveData() {
+    tcf_err_t presult;
+
+    // UnsealEnclaveData will resize appropriately
+    StringArray public_enclave_data(0);
+
+    SignupDataKME signup_data;
+    presult = signup_data.UnsealEnclaveData(
+        public_enclave_data);
+    ThrowTCFError(presult);
+
+    // Parse the json and save the verifying and encryption keys
+    std::string verifying_key;
+    std::string encryption_key;
+    std::string encryption_key_signature;
+
+    presult = SignupInfo::DeserializePublicEnclaveData(
+        public_enclave_data.str(),
+        verifying_key,
+        encryption_key,
+        encryption_key_signature);
+    ThrowTCFError(presult);
+
+    std::map<std::string, std::string> result;
+    result["verifying_key"] = verifying_key;
+    result["encryption_key"] = encryption_key;
+    result["encryption_key_signature"] = encryption_key_signature;
```

Comment:
Please clear the encryption_key variable explicitly after use.

Reply:
These keys are public keys and this is untrusted part of the code which anyway doesn't run inside enclave.